R/module_rds.R

```diff
@@ -0,0 +1,118 @@
+#' rds tab UI
+#' Module to create a tabset panel to allow the download of the rds file
+#' @param id ID for shiny module namespacing
+#' @noRd
+rdsUI = function(id) {
+  ns = shiny::NS(id)
+  # Plots tab panel
+  shiny::tabPanel("RDS Files",
+                  # drop down menu to select plot
+                  shiny::br(),
+                  shiny::uiOutput(ns("choose_rds")),
+                  # display plot
+                  shiny::wellPanel(
+                    shiny::fluidRow(shiny::column(
+                      12,
+                      align = "center",
+                      shiny::uiOutput(ns("display_rds")),
+                      style = "height:400px;")),
+                    style = "background: white"
+                  ),
+                  # download button to download current plot
+                  shiny::br(),
+                  shiny::fluidRow(
+                    shiny::column(12,
+                                  align = "center",
+                                  shiny::uiOutput(ns("download_rds_button")) #nolint
+                    )
+                  )
+  )
+}
+
+#' rds tab Server
+#' @param id ID for shiny module namespacing
+#' @param cluster_choice which cluster to display the data for
+#' @noRd
+rdsServer = function(id, cluster_choice) {
+  shiny::moduleServer(id, function(input, output, session) {
+    ns = session$ns
+
+    # all available rds
+    all_files = shiny::reactive({
+      return(get_all_files(cluster_choice()))
+    })
+
+    # drop down for plots
+    output$choose_rds = shiny::renderUI({
+      all_rds = all_files() %>%
+        dplyr::filter(.data$filetype %in% c("rds", "RDS")) %>%
+        dplyr::pull(.data$value)
+      rds_names = stringr::str_to_title(
+        stringr::str_replace_all(
+          gsub("\\..*", "", all_rds), "_", " "))
+      names(all_rds) = rds_names
+      shiny::selectInput(ns("rds_type"),
+                         label = "Select RDS file:",
+                         choices = all_rds)
+    })
```

Comment:
There's some very similar code in the plots and tables modules.
Could you extract the inner code into a function and use that in module_[rds/plots/tables].

```r
filter_by_filetype = function(filenames, filetypes) {
  matching_files = filenames %>%
        dplyr::filter(.data$filetype %in% .env[["filetypes"]]) %>%
        dplyr::pull(.data$value)

 file_names = stringr::str_to_title(
        stringr::str_replace_all(
          gsub("\\..*", "", matching_files), "_", " "))

  names(matching_files) = file_names
  matching_files
}
```

Reply:
Thanks - I've added this function, and updated the code inside the modules.